.circleci/config.yml

```diff
@@ -65,20 +67,63 @@ jobs:
       - run:
           name: Test
           command: bash -i -c 'npx cypress run --record --config numTestsKeptInMemory=1 --spec cypress/integration/group3/**/*'
+  upload_to_s3:
+    working_directory: ~/repo
+    docker:
+      - image: circleci/python:2.7
+    parameters:
+      aws_bucket:
+        type: string
+        default: "${AWS_BUCKET}"
+    steps:
+      - when:
+          condition: <<parameters.aws_bucket>>
+          steps:
+            - attach_workspace:
+                at: .
+            - aws-s3/copy:
+                from: dist
+                to: 's3://${AWS_BUCKET}/${CIRCLE_TAG}-$(echo $CIRCLE_SHA1 | cut -c -7)'
+                arguments: '--recursive'
+
 workflows:
   version: 2
   everything:
     jobs:
-      - lint_unit_test_coverage
+      # Add the tags filter to all jobs so they will run before upload_to_s3
+      - lint_unit_test_coverage:
+          filters:
+            tags:
+              only: /.*/
       - integration_test_1:
+          filters:
+            tags:
+              only: /.*/
```

Comment:
~Does this limit integration tests to tags only as well?~

Actually, I see tests above as part of jobs standalone and here. Will that lead to the tests running twice? 
What's the interaction between the tests in the workflows root and in the jobs root?

Reply:
See [this doc](https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag), in particular the second code snippet in that section.

My understanding from that is that by default, jobs are only run for branch commits. I've added an upload_to_s3 job which is configured to only run for tags. Since upload_to_s3 requires a build, that means I need to tell the build jobs to also run for tags, hence this section.

For a branch commit, only build jobs (integration_test_1, etc.) will run.

For a tag, build jobs and upload_to_s3 will run.